src/ray/core_worker/task_interface.cc

```diff
@@ -25,17 +26,64 @@ ActorHandle::ActorHandle(
   inner_.set_is_direct_call(is_direct_call);
 }
 
-ActorHandle::ActorHandle(const ActorHandle &other)
-    : inner_(other.inner_), new_actor_handles_(other.new_actor_handles_) {}
+ActorHandle::ActorHandle(ActorHandle &parent, bool in_band) {
+  std::unique_lock<std::mutex> guard(parent.mutex_);
+  inner_ = parent.inner_;
+  class ActorHandleID new_actor_handle_id;
+  if (in_band) {
+    new_actor_handle_id = ComputeForkedActorHandleId(
+        ActorHandleID::FromBinary(parent.inner_.actor_handle_id()),
+        parent.inner_.num_forks());
+    // Notify the backend to expect this new actor handle. The backend will
+    // not release the cursor for any new handles until the first task for
+    // each of the new handles is submitted.
+    // NOTE(swang): There is currently no garbage collection for actor
+    // handles until the actor itself is removed.
+    new_actor_handles_.push_back(new_actor_handle_id);
```

Comment:
this looks like a bug, new handle id should be put into the parent's `new_actor_handles_`, not the one for itself

Reply:
could you also add a test case to cover this logic?

Reply:
I think you were looking at an old revision. Will add test.